mmv1/third_party/terraform/website/docs/guides/google_project_service.html.markdown

```diff
@@ -0,0 +1,107 @@
+---
+page_title: "User Guide for google_project_service"
+description: |-
+  An advanced user guide for the google_project_service resource
+---
+
+# User Guide - google_project_service
+
+## Enabling Multiple Services in a Config
+
+Users may want to activate many services simultaneously within a single Terraform config. The `google_project_service.service` field only supports a single value, but Terraform itself provides list iteration with [for_each](https://developer.hashicorp.com/terraform/language/meta-arguments/for_each).
+A simple example:
+
+```
+variable "services" {
+  type = list(string)
+}
+
+resource "google_project_service" "services" {
+  for_each = toset(var.services)
+  project = "my-project"
+  service = each.value
+}
+```
+
+For a more robust example, Google recommends this [project_services module](https://github.com/terraform-google-modules/terraform-google-project-factory/tree/master/modules/project_services). This module aims to simplify the configuration needed for service enablement to [this example](https://github.com/terraform-google-modules/terraform-google-project-factory/tree/master/modules/project_services#example-usage).
+
+## Request Rate Errors
+
+The service management API called by the google_project_service resource has rate limits on the requesting user account if no quota project is set. When using the default (credentials) quota project or a billing_project users enabling many GCP services simultaneously may experience rate limit errors from the API. `google_project_service` supports batching multiple resources into smaller requests, see the `batching` [reference documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference.html#batching).
```

Comment:
There's a bit of whiplash here going from "...if no quota project is set." to "When using the default (credentials) quota project..." as you're going from saying there's no quota project and then saying there's a default one! Here's a potential rewording:

```suggestion
The service management API called by the google_project_service resource uses request rate quota on the project of the account used to call the API (i.e. against the Terraform credentials) by default. That project (or a fixed `billing_project`) may exceed your request rate quota in larger configurations. `google_project_service` batches multiple changes into single requests when possible, see the `batching` [reference documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference.html#batching) for details.
```